gap/digraph.gi

```diff
@@ -429,6 +444,98 @@ DigraphCons);
 InstallMethod(Digraph, "for a list, list, and list", [IsList, IsList, IsList],
 {dom, src, ran} -> DigraphCons(IsImmutableDigraph, dom, src, ran));
 
+InstallMethod(Digraph, "for a string naming a digraph", [IsString],
+function(name)
+  # edge case to avoid interfering with Digraph([])
+  if name = "" then
+    TryNextMethod();
+  fi;
+
+  # standardise string format to search database
+  name := LowercaseString(name);
+  RemoveCharacters(name, " \n\t\r");
+
+  # load database if not already done
+  DIGRAPHS_LoadNamedDiSparse6Strings();
+
+  if not name in RecNames(DIGRAPHS_NamedDiSparse6Strings) then
+    ErrorNoReturn("Named digraph not found. Please check argument 'name',\n",
+                  "or view list of available digraphs with names containing\n",
+                  "string s using ListNamedDigraphs(s).");
+  fi;
+  return DigraphFromDiSparse6String(DIGRAPHS_NamedDiSparse6Strings.(name));
+end);
+
+InstallMethod(ListNamedDigraphs,
+"for a partially completed string and a pos int",
+[IsString, IsPosInt],
+function(s, level)
+  local l, cands, out, c;
+  # standardise request
+  s := LowercaseString(s);
+  RemoveCharacters(s, " \n\t\r");
+  l := Length(s);
+
+  # load database if not already done
+  DIGRAPHS_LoadNamedDiSparse6Strings();
+
+  # retrieve candidates
+  cands := RecNames(DIGRAPHS_NamedDiSparse6Strings);
+  if l = 0 then
+    return cands;
+  fi;
+
+  out := [];
+
+  # print warning if level higher than ones here that have methods
+  if level > 3 then
+    Info(InfoWarning, 1, "ListNamedDigraphs: second argument <level> is");
+    Info(InfoWarning, 1, "greater than level of greatest flexibility.");
+    Info(InfoWarning, 1, "Using <level> = 3 instead.");
+    level := 3;
+  fi;
+
+  # IF LEVEL = 1 (PREFIX SEARCH):
+  # add to out-list all valid completions of the request s
+  if level = 1 then
+    for c in cands do
+      if Length(c) >= l and c{[1 .. l]} = s then
+        Add(out, c);
+      fi;
+    od;
+  fi;
+
+  # IF LEVEL = 2 (SUBSTRING SEARCH):
+  # add to out-list all strings that contain request s
+  if level = 2 then
+    for c in cands do
+      if PositionSublist(c, s) <> fail then
+        Add(out, c);
+      fi;
+    od;
+  fi;
+
+  # IF LEVEL = 3 (ALPHANUM-ONLY SUBSTRING SEARCH):
+  # add to out-list all strings that when stripped to contain only alphanumeric
+  # characters, contain the resquest s as a substring
+  if level = 3 then
+    s := Filtered(s, x -> IsDigitChar(x) or IsAlphaChar(x));
+    for c in cands do
+      if PositionSublist(Filtered(c, x -> IsDigitChar(x) or IsAlphaChar(x)), s)
+          <> fail then
+        Add(out, c);
+      fi;
+    od;
+  fi;
+
+  return out;
```

Comment:
This could be a much shorter way of doing this (with whatever comments you want to add)
```suggestion
  if level = 1 then
    func := c -> Length(c) >= l and c{[1 .. l]} = s;
  elif level = 2 then
    func := c -> PositionSublist(c, s) <> fail;
  else
    s := Filtered(s, x -> IsDigitChar(x) or IsAlphaChar(x));
    func := c -> PositionSublist(Filtered(c, x -> IsDigitChar(x) or IsAlphaChar(x)), s) <> fail;
  fi;
  return Filtered(cands, func);
```